iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -411,10 +414,12 @@ public boolean canSetColStatistics(org.apache.hadoop.hive.ql.metadata.Table hmsT
 
   @Override
   public boolean setColStatistics(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
-      List<ColumnStatistics> colStats) {
+      List<ColumnStatistics> colStats, ColumnStatsDesc columnStatsDesc) {
     Table tbl = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
     String snapshotId = String.format("%s-STATS-%d", tbl.name(), tbl.currentSnapshot().snapshotId());
-    invalidateStats(getStatsPath(tbl));
+    if (!checkAndInvalidateStats(tbl)) {
+      checkAndMergeStats(colStats.get(0), tbl, hmsTable, columnStatsDesc);
```

Comment:
shouldn't we always merge the stats?

Reply:
if it is an `analyze  <table> compute statistics for column`, we should not merge the stats but overwrite them with the complete stats. So for analyze table command we should invalidate the stats, here it was done by deleting the older stats.